cloudpickle/cloudpickle.py

```diff
@@ -1127,6 +1141,10 @@ def _class_setstate(obj, state):
         if attrname == "_abc_impl":
             registry = attr
         else:
+            # Note: attribute names are automatically interned in cpython. This means that to get
+            # determinist pickling in subprocess, we need to make sure that the dynamic function names
+            # are also interned.
```

Comment:
```suggestion
            # are also interned since the Pickler's memoizer relies on physical object
            # identity to break cycles in the reference graph of the object being
            # serialized.
```